assets/onboarding/data/reducer.js

```diff
@@ -29,6 +29,34 @@ const DEFAULT_STATE = {
 	},
 };
 
+/**
+ * @typedef  {Object} Feature
+ * @property {string} slug    Feature slug.
+ * @property {string} status  Feature status.
+ * @property {Object} error   Feature error.
+ */
+/**
+ * Remove error and status from selected options.
+ *
+ * @param {string[]}  selected Feature slugs.
+ * @param {Feature[]} options  Options.
+ *
+ * @return {Feature[]} Updated options.
+ */
+const removeErrorFromSelected = ( selected, options ) =>
```

Comment:
We could separate the installation status from the extension info in the features data, which I think would simplify a lot of things in updating the state:
```
{
  selected: [ 'slugs' ],
  status: { slug: { installed: true } },
  options: [ ... ] // Extension info
}
```
A reducer can update just the status both when polling and on retry, with just an action instead of extra hoops with a resolver and invalidation.  For some optimization, polling can fetch just the status from a `GET features-installation` endpoint. We could even just treat `features-installation` as a top-level step with just the status data.

Might be too much work to change, so not critical if you think it's not worth it.

Reply:
I like the idea, but yes, it's a big change. Maybe we could leave it for a future refactory, if you agree :)